pkg/controller/workload/job/job_controller.go

```diff
@@ -433,27 +482,49 @@ func ConstructWorkloadFor(ctx context.Context, client client.Client,
 }
 
 func appendFinishedConditionIfNotExists(conds []metav1.Condition, jobStatus batchv1.JobConditionType) ([]metav1.Condition, bool) {
+	return appendConditionIfNotExists(conds, generateFinishedCondition(jobStatus))
+}
+
+func generateFinishedCondition(jobStatus batchv1.JobConditionType) metav1.Condition {
+	message := "Job finished successfully"
+	if jobStatus == batchv1.JobFailed {
+		message = "Job failed"
+	}
+	return metav1.Condition{
+		Type:    kueue.WorkloadFinished,
+		Status:  metav1.ConditionTrue,
+		Reason:  "JobFinished",
+		Message: message,
+	}
+}
+
+func appendPodsReadyConditionIfNotExists(conds []metav1.Condition, jobStatus *batchv1.JobStatus) ([]metav1.Condition, bool) {
+	return appendConditionIfNotExists(conds, generatePodsReadyCondition(jobStatus))
+}
+
+func generatePodsReadyCondition(jobStatus *batchv1.JobStatus) metav1.Condition {
+	podsReadyCount := podsReadyCount(jobStatus)
+	message := fmt.Sprintf("All pods are ready or succeeded. Ready: %v, succeeded: %v.", podsReadyCount, jobStatus.Succeeded)
+	return metav1.Condition{
+		Type:    kueue.WorkloadPodsReady,
+		Status:  metav1.ConditionTrue,
+		Reason:  "PodsReady",
+		Message: message,
+	}
+}
+
+func appendConditionIfNotExists(conds []metav1.Condition, condition metav1.Condition) ([]metav1.Condition, bool) {
```

Comment:
you can use `k8s.io/apimachinery/pkg/api/meta.SetStatusCondition`

Reply:
Good idea, used. Also applied it to add the `WorkloadFinished` condition for consistency.